implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/DynatraceConfig.java

```diff
@@ -37,21 +40,27 @@ default String prefix() {
     }
 
     default String apiToken() {
-        return getSecret(this, "apiToken").required().get();
+        if (apiVersion() == DynatraceApiVersion.V1) {
+            return getSecret(this, "apiToken").required().get();
+        }
+        return getSecret(this, "apiToken").orElse("NO_TOKEN_SET");
```

Comment:
Would it be possible to just use `null` here instead of some magic string? Otherwise I'd find the empty string cleaner if making it nullable would constitute a breaking change.

Reply:
due to the NonNull annotations we solved this using empty strings.